pelican/plugins/show_source/show_source.py

```diff
@@ -29,20 +29,13 @@ def link_source_files(generator):
 
     # Work on each item
     for post in documents:
+        # condition to show a post
         if not (
-            "SHOW_SOURCE_ON_SIDEBAR" in generator.settings
-            or "SHOW_SOURCE_IN_SECTION" in generator.settings
+            generator.settings.get("SHOW_SOURCE_ALL_POSTS")
+            or post.metadata.get("show_source")
         ):
             logger.debug("show_source: sources not shown, aborting plugin")
-            return
-
-        # Only try this when specified in metadata or SHOW_SOURCE_ALL_POSTS
-        # override is present in settings
-        if "SHOW_SOURCE_ALL_POSTS" not in generator.settings:
```

Comment:
Can you explain this removal?

Reply:
This fixes the logic for all the test cases (check the output of the py test to get a better feeling for it): it covers all the cases for the flags now.

Let's use a tuple for these values:
  (SHOW_SOURCE_ALL_POSTS, SHOW_SOURCE_ON_SIDEBAR, SHOW_SOURCE_IN_SECTION)

**case (0, 0, 1)**
(eg. SHOW_SOURCE_ALL_POSTS=False, SHOW_SOURCE_ON_SIDEBAR=False, SHOW_SOURCE_IN_SECTION=True)
We expect the output to show the "show source" inline to the articles 1 & 3 (they both have Show_source: True in the meta) but not in 2.
**case (0, 1, )**
in the "side bar" the show source should show again on the page 1 &3 but non on 2.

And so on.. there are 16 cases for each flag (including the SHOW_SOURCE_AUTOEXT).

BTW _if "SHOW_SOURCE_ALL_POSTS" not in generator.settings_ -> that will evaluate to True even if the flag is set to False.